Client/src/main/java/de/baleipzig/pdfextraction/client/controller/ActionController.java

```diff
@@ -85,6 +96,69 @@ private void runActionButtonOnAction() {
          */
     }
 
+    private void toggleActionButtonVisible(boolean isVisible) {
+        runActionButton.setVisible(isVisible);
+        progress.setVisible(!isVisible);
+    }
+
+    private void onCompleteGetAllNames(String name) {
+
+        FXMLLoader loader = new FXMLLoader(new ActionItem().getFXML());
+        try {
+            Node item = loader.load();
+            buildItem(loader, name);
+            contentPane.getChildren().add(item);
+        } catch (IOException e) {
+            LoggerFactory.getLogger(getClass())
+                    .atError()
+                    .setCause(e)
+                    .log("Exception occurred while create item");
+
+            AlertUtils.showErrorAlert(e);
+        }
+    }
+
+    private void buildItem(FXMLLoader loader, String name) {
+
+        ActionItemController actionItemController = loader.getController();
+        actionItemController.actionLabel.setText(name);
+        actionItemController.itemPane.setOnMouseClicked(event -> {
+            if (actionItemController.selectActionCircle.getStrokeWidth() == STROKE_WIDTH_CIRCLE) {
+                GridPane gridPaneContent = checkOneActionIsSelected(Arrays.asList(contentPane, contentPaneOthers));
+                if(gridPaneContent != null){
+                    HBox hBoxCircleOld = (HBox) getActionItemById(gridPaneContent, "hBoxCircle");
+                    Circle selectActionCircleOld = (Circle) getActionItemById(hBoxCircleOld, "selectActionCircle");
+                    selectActionCircleOld.setStrokeWidth(STROKE_WIDTH_CIRCLE);
+                    selectActionCircleOld.setFill(Paint.valueOf("#ffffff00"));
+                }
+                actionItemController.selectActionCircle.setStrokeWidth(0);
+                actionItemController.selectActionCircle.setFill(done);
+            } else {
+                actionItemController.selectActionCircle.setStrokeWidth(STROKE_WIDTH_CIRCLE);
+                actionItemController.selectActionCircle.setFill(Paint.valueOf("#ffffff00"));
+            }
+        });
+
+    }
+
+    private GridPane checkOneActionIsSelected(List<JFXMasonryPane> contentPanes) {
+
+        for (JFXMasonryPane contentPane : contentPanes) {
+            for (int i = 0; i < contentPane.getChildren().size(); i++) {
+
+                GridPane gridPaneContent = (GridPane) getActionItemById(contentPane.getChildren().get(i), "gridPaneContent");
+                HBox hBoxCircle = (HBox) getActionItemById(gridPaneContent, "hBoxCircle");
+                Circle selectActionCircle = (Circle) getActionItemById(hBoxCircle, "selectActionCircle");
+
+                if (isItemSelected(selectActionCircle)) {
+                    return gridPaneContent;
+                }
```

Comment:
Was ist falsch mit einer normalen for-Schleife?

Reply:
Was ist eine normale for-Schleife?